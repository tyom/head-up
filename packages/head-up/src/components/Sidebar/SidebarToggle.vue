<template>
  <div class="SidebarToggle">
    <button class="toggle" @click="$emit('toggle')">
      <v-icon :name="toggleIconName"/>
    </button>
  </div>
</template>

<script>
export default {
  props: {
    toggled: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    toggleIconName() {
      return this.toggled ? 'chevron-left' : 'ellipsis-v';
    },
  },
};
</script>

<style scoped>
.SidebarToggle {
  overflow: hidden;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  z-index: 2;
  box-shadow: 1px 0 1px rgba(#000, 0.5), inset -1px 0 rgba(#fff, 0.1);
}

.toggle {
  width: 1.6rem;
  height: 100%;
  font-size: 1rem;
  border: 0;
  color: inherit;
  outline: 0;
  background-color: transparent;
  transition: background-color 0.1s;

  &:hover {
    color: #fff;
    background-color: rgba(#000, 0.2);
  }

  &:active {
    background-color: #142a59;
    transform: scaleY(1.2);
  }

  &.focus-visible {
    box-shadow: inset 0 0 0 1px rgba(#fff, 0.3);
  }
}
</style>
