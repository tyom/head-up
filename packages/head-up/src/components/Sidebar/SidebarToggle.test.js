import { shallowMount } from '@vue/test-utils';
import SidebarToggle from './SidebarToggle';

test('render untoggled', () => {
  const wrapper = shallowMount(SidebarToggle, {
    propsData: {
      toggled: false,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('render toggled', () => {
  const wrapper = shallowMount(SidebarToggle, {
    propsData: {
      toggled: true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('responds to actions', () => {
  const wrapper = shallowMount(SidebarToggle);

  wrapper.find('button').trigger('click');
  expect(wrapper.emitted('toggle')[0]).toBeTruthy();
});
