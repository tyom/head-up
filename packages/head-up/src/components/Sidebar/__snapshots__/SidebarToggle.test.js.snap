// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render toggled 1`] = `
<div class="SidebarToggle"><button class="toggle">
    <v-icon-stub name="chevron-left"></v-icon-stub>
  </button></div>
`;

exports[`render untoggled 1`] = `
<div class="SidebarToggle"><button class="toggle">
    <v-icon-stub name="ellipsis-v"></v-icon-stub>
  </button></div>
`;
